invenio_sipstore/archivers.py

```diff
@@ -0,0 +1,153 @@
+        bagit_file_info = self.create_bagit_file()
+        baginfo_file_info = self.create_baginfo_file()
+
+        self.create_tagmanifest([manifest_file_info,
+                                 bagit_file_info,
+                                 baginfo_file_info, ])
+
+    def copy_files(self, files, progress_callback):
+        """Copy the files inside the new storage."""
+        result = {}
+
+        total_size = sum([file.file.size for file in files])
+        copied = 0
+
+        for file in files:
+            src_fs, path = opener.parse(file.file.uri)
+            calculated_filename = os.path.join('data', 'files', file.filepath)
+            self.create_directories(calculated_filename)
+
+            copyfile(src_fs, file.file.storage().filename,
+                     self.fs, calculated_filename)
+            file_size = file.file.size
+            result[calculated_filename] = {
+                'size': file_size,
+                'checksum': file.file.checksum,
+            }
+            copied += file_size
+            progress_callback(file_size, total_size)
+        return result
+
+    def autogenerate_tags(self, files_info):
+        """Generate the automatic tags."""
+        self.tags['Bagging-Date'] = datetime.now().strftime(
+            "%Y-%m-%d_%H:%M:%S:%f")
+        self.tags['Payload-Oxum'] = '{0}.{1}'.format(
+            sum([f['size'] for f in files_info.values()]), len(files_info))
+
+    def create_manifest_file(self, files_info):
+        """Create the manifest file spcifying the checksum of the files."""
+        content_items = ('{0} {1}'.format(get_checksum(c['checksum']), f)
+                         for f, c in files_info.items())
+        content = '\n'.join(content_items)
+        return self.save_file(['manifest-md5.txt'], content)
+
+    def create_bagit_file(self):
+        """Create the bagit file which specify the version and encoding."""
+        content_items = [
+            'BagIt-Version: 0.97',
+            'Tag-File-Character-Encoding: UTF-8',
+        ]
+        content = '\n'.join(content_items)
+        return self.save_file(['bagit.txt'], content)
+
+    def create_baginfo_file(self):
+        """Create the baginfo file using the tags."""
+        content_items = ('{0}: {1}'.format(k, v)
+                         for k, v in self.tags.items())
+        content = '\n'.join(content_items)
+        return self.save_file(['bag-info.txt'], content)
+
+    def create_tagmanifest(self, files_info):
+        """Create the tagmanifest file using the files info."""
+        content_items = ('{0} {1}'.format(get_checksum(fileinfo['checksum']),
+                                          fileinfo['filename'])
+                         for fileinfo in filter(None, files_info))
+        content = '\n'.join(content_items)
+        return self.save_file(['tagmanifest-md5.txt'], content)
+
+    def save_file(self, filename, content):
+        """Save the content inside the file."""
```

Comment:
Perhaps add a warning in the documentation here that this is only to write small files.